static/app/components/core/button/useButtonFunctionality.tsx

```diff
@@ -0,0 +1,74 @@
+import HookStore from 'sentry/stores/hookStore';
+
+import {type LinkButtonProps} from './linkButton';
+import {type ButtonProps} from './';
```

Comment:
I know the import is “just types”, but still, importing from the `index` file here, where the `index` file also imports `useButtonFunctionality` is a circularity we should try to avoid. Since you created a `types.tsx` file - why not move the `ButtonProps` there, too ?

Or, alternatively, move the `type ButtonProps` into `useButtonFunctionality`. I think we should avoid generically named files like `types.tsx` or `utils.tsx` because it will likely act as a magnet for all sorts of unrelated things. Probably fine in the scope of `core/button`, but we also have this on the top-level and it can quickly become a pile of unrelated groupings.

Reply:
> think we should avoid generically named files like types.tsx or utils.tsx 

I generally do agree with this, I sort of think a `types.tsx` file is OK, but ONLY within the context of a indivudual module. I don't think we should have huge global types.

---

Let me see if I can move the types around a bit more